src/main/java/org/commcare/formplayer/services/MediaMetaDataService.java

```diff
@@ -39,21 +39,11 @@ public MediaMetadataRecord findByFileId(String id) {
         return record.get();
     }
 
-    @Cacheable
-    public MediaMetadataRecord getMetadataById(String id) {
-        Optional<MediaMetadataRecord> record = mediaMetaDataRepo.findById(id);
-        if (!record.isPresent()) {
-            throw new MediaMetaDataNotFoundException(id);
-        }
-        return record.get();
-    }
-
-    @CachePut(key = "#mediaMetadataRecord.id")
+    @CachePut(key = "#mediaMetadataRecord.fileId")
     public MediaMetadataRecord saveMediaMetaData(MediaMetadataRecord mediaMetadataRecord) {
         return mediaMetaDataRepo.save(mediaMetadataRecord);
     }
 
-    @CacheEvict(key = "#id")
     public void deleteMetaDataById(String id) {
```

Comment:
Calling any delete method sshould also evict the corresponding record from cache as otherwise read will return the cached record even when a record has been deleted from DB.

Also do we really need to implement delete by id ? Instead can we implement and use `deleteByFileId` since we are doing other if it's simpler to do ?

Reply:
Thanks. I should have removed that when I switched over to using `fileId`.

Reply:
> Calling any delete method should also evict the corresponding record from cache

Thanks, that makes sense.